src/v/raft/consensus.cc

```diff
@@ -266,6 +266,10 @@ consensus::success_reply consensus::update_follower_index(
         // we are already recovering, if follower dirty log index moved back
         // from some reason (i.e. truncation, data loss, trigger recovery)
         if (idx.last_dirty_log_index > reply.last_dirty_log_index) {
+            // update follower state to allow recovery of follower with
+            // missing entries
+            idx.last_dirty_log_index = reply.last_dirty_log_index;
```

Comment:
ah lgtm. so you don't need the `details::next_offset()` for any of these?

Reply:
for those two we do not need next.